tools/codegen/model.py

```diff
@@ -1496,3 +1503,40 @@ def parse_returns(return_decl: str) -> Tuple[Return, ...]:
     if return_decl[0] == '(' and return_decl[-1] == ')':
         return_decl = return_decl[1:-1]
     return tuple(Return.parse(arg) for arg in return_decl.split(', '))
+
+@dataclass(frozen=True)
+class PrecomputedElement:
+    name: str
+    ty: str
+
+    @staticmethod
+    def build(src: str) -> 'PrecomputedElement':
+        ty, name = src.split(" ")
+        return PrecomputedElement(name=name, ty=ty)
+
+    def decl(self) -> str:
+        return f"{self.ty} {self.name}"
+
+@dataclass(frozen=True)
+class Precompute:
+    elements: List[Argument]
```

Comment:
s/Argument/PrecomputedElement/ right?

Reply:
Yes! I tried to use `Argument` first to avoid introducing another class, but got stuck because `Argument` parses JIT types, not C++ types.